rs/replicated_state/src/canister_state/queues/queue.rs

```diff
@@ -239,12 +203,12 @@ impl CanisterQueue {
         }
         debug_assert_eq!(Ok(()), self.check_invariants());
 
-        Some(item)
+        Some(reference)
     }
 
-    /// Returns the next item in the queue; or `None` if the queue is empty.
-    pub(super) fn peek(&self) -> Option<&CanisterQueueItem> {
-        self.queue.front()
+    /// Returns the next reference in the queue; or `None` if the queue is empty.
+    pub(super) fn peek(&self) -> Option<message_pool::Id> {
+        self.queue.front().cloned()
```

Comment:
We now always clone here because the ID has the same size as a pointer would have anyways, right?

Reply:
Right. And while I'm not 100% sure of this, using a pointer to an integer is bound to be less efficient than using the integer directly (which is not necessarily true for anything larger).